From 54886df43b82ab5e4c2e2ef2c1be4d66d467d8f9 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Sun, 16 Nov 2014 10:39:01 +0000
Subject: [PATCH] MythUIHelper: use the theme search paths to find a
 qtlook.txt

This allows a theme to fallback to any base themes qtlook.txt file.

We now also warn if one wasn't found.
---
 mythtv/libs/libmythui/mythuihelper.cpp |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/mythtv/libs/libmythui/mythuihelper.cpp b/mythtv/libs/libmythui/mythuihelper.cpp
index 245bbb7..b9c791a 100644
--- a/mythtv/libs/libmythui/mythuihelper.cpp
+++ b/mythtv/libs/libmythui/mythuihelper.cpp
@@ -500,11 +500,14 @@ void MythUIHelper::LoadQtConfig(void)
     // Recalculate GUI dimensions
     d->StoreGUIsettings();
 
-    d->m_searchPaths.clear();
     d->m_themepathname = themedir + '/';
+    d->m_searchPaths.clear();
+
+    QString qtlook = "qtlook.txt";
+    if (!FindThemeFile(qtlook))
+        LOG(VB_GENERAL, LOG_ERR, LOC + "Failed to find any qtlook.txt in the theme search path");
 
-    themedir += "/qtlook.txt";
-    d->m_qtThemeSettings->ReadSettings(themedir);
+    d->m_qtThemeSettings->ReadSettings(qtlook);
     d->m_themeloaded = false;
 
     themename = GetMythDB()->GetSetting("MenuTheme", "defaultmenu");
-- 
1.7.10.2

